File: solutions/javascript/src/structure/fixed/StaticSetADT.js
class StaticSetADT {
  constructor(capacity) {
    this.capacity = capacity;
    this.items = new Array(capacity);
    this.size = 0;
  }

  exist(value) {
    for (let i = 0; i < this.size; i++) {
      if (this.items[i] === value) return true;
    }
    return false;
  }

  choose() {
    if (this.isEmpty()) {
      throw new Error("El conjunto está vacío");
    }
    const index = Math.floor(Math.random() * this.size);
    return this.items[index];
  }

  add(value) {
    if (this.size >= this.capacity) {
      throw new Error("El conjunto ha alcanzado su capacidad máxima");
    }
    if (!this.exist(value)) {
      this.items[this.size++] = value;
    }
  }

  remove(value) {
    const index = this.items.indexOf(value);
    if (index !== -1) {
      for (let i = index; i < this.size - 1; i++) {
        this.items[i] = this.items[i + 1];
      }
      this.size--;
    }
  }

  isEmpty() {
    return this.size === 0;
  }
}

File: solutions/python/structure/fixed/StaticSetADT.py
import random

class StaticSetADT:
    def __init__(self, capacity):
        self.capacity = capacity
        self.elements = [None] * capacity
        self.size = 0

    def exist(self, value):
        for i in range(self.size):
            if self.elements[i] == value:
                return True
        return False

    def choose(self):
        if self.isEmpty():
            raise Exception("No se puede elegir de un conjunto vacío")
        return random.choice(self.elements[:self.size])

    def add(self, value):
        if self.size >= self.capacity:
            raise Exception("Capacidad máxima alcanzada")
        if self.exist(value):
            return
        self.elements[self.size] = value
        self.size += 1

    def remove(self, value):
        for i in range(self.size):
            if self.elements[i] == value:
                for j in range(i, self.size - 1):
                    self.elements[j] = self.elements[j + 1]
                self.elements[self.size - 1] = None
                self.size -= 1
                return

    def isEmpty(self):
        return self.size == 0
